.Lib9c.Tests/Action/BuyMultipleTest.cs

```diff
@@ -0,0 +1,670 @@
+            _buyerAvatarAddress = new PrivateKey().ToAddress();
+            var rankingMapAddress = new PrivateKey().ToAddress();
+            _buyerAvatarState = new AvatarState(
+                _buyerAvatarAddress,
+                _buyerAgentAddress,
+                0,
+                _tableSheets.GetAvatarSheets(),
+                new GameConfigState(),
+                rankingMapAddress)
+            {
+                worldInformation = new WorldInformation(
+                    0,
+                    _tableSheets.WorldSheet,
+                    GameConfig.RequireClearedStageLevel.ActionsInShop),
+            };
+            buyerAgentState.avatarAddresses[0] = _buyerAvatarAddress;
+
+            var shopState = new ShopState();
+
+            _initialState = _initialState
+                .SetState(GoldCurrencyState.Address, _goldCurrencyState.Serialize())
+                .SetState(Addresses.Shop, shopState.Serialize())
+                .SetState(_buyerAgentAddress, buyerAgentState.Serialize())
+                .SetState(_buyerAvatarAddress, _buyerAvatarState.Serialize())
+                .MintAsset(_buyerAgentAddress, _goldCurrencyState.Currency * 100);
+        }
+
+        public static IEnumerable<object[]> GetExecuteMemberData()
+        {
+            yield return new object[]
+            {
+                new ShopItemData()
+                {
+                    ItemType = ItemType.Equipment,
+                    ItemId = Guid.NewGuid(),
+                    SellerAgentAddress = new PrivateKey().ToAddress(),
+                    SellerAvatarAddress = new PrivateKey().ToAddress(),
+                    RequiredBlockIndex = Sell.ExpiredBlockIndex,
+                    Buy = true,
+                    Price = 10,
+                    ContainsInInventory = true,
+                },
+                new ShopItemData()
+                {
+                    ItemType = ItemType.Costume,
+                    ItemId = Guid.NewGuid(),
+                    SellerAgentAddress = new PrivateKey().ToAddress(),
+                    SellerAvatarAddress = new PrivateKey().ToAddress(),
+                    RequiredBlockIndex = Sell.ExpiredBlockIndex,
+                    Buy = false,
+                    Price = 20,
+                    ContainsInInventory = true,
+                },
+                new ShopItemData()
+                {
+                    ItemType = ItemType.Costume,
+                    ItemId = Guid.NewGuid(),
+                    SellerAgentAddress = new PrivateKey().ToAddress(),
+                    SellerAvatarAddress = new PrivateKey().ToAddress(),
+                    RequiredBlockIndex = 0,
+                    Buy = true,
+                    Price = 30,
+                    ContainsInInventory = false,
+                },
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetExecuteMemberData))]
+        public void Execute(params ShopItemData[] productDatas)
```

Comment:
If you use `Theory`, it seems that you will run the test three times while going through the `ShopItemData` one by one, this is not correct test purchasing multiple items in `BuyMultiple`.

Reply:
The test runs just once with three `ShopItemData`. I think the test case is just not enough, so i will add some more.